From 8e15369f7e7d5cb97fb8313cd611a165c9267c96 Mon Sep 17 00:00:00 2001
From: Alexander Stein <alexander.stein@tq-group.com>
Date: Wed, 6 Jul 2022 12:55:51 +0200
Subject: [PATCH 1/1] glupload: Do not hard-code RGBA format for sink pad

This prevents proper pipeline negotiation when the src pad has a non-RGB
or non-YUV format.

Signed-off-by: Alexander Stein <alexander.stein@tq-group.com>
---
 gst-libs/gst/gl/gstglupload.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/gst-libs/gst/gl/gstglupload.c b/gst-libs/gst/gl/gstglupload.c
index 78b65c1cc..0556d4133 100644
--- a/gst-libs/gst/gl/gstglupload.c
+++ b/gst-libs/gst/gl/gstglupload.c
@@ -560,7 +560,6 @@ _dma_buf_upload_transform_caps (gpointer impl, GstGLContext * context,
         _set_caps_features_with_passthrough (caps,
         GST_CAPS_FEATURE_MEMORY_GL_MEMORY, passthrough);
 
-    gst_caps_set_simple (ret, "format", G_TYPE_STRING, "RGBA", NULL);
     tmp = _caps_intersect_texture_target (ret, 1 << GST_GL_TEXTURE_TARGET_2D);
     gst_caps_unref (ret);
     ret = tmp;
-- 
2.25.1

